src/backend/distributed/commands/table.c

```diff
@@ -1886,6 +1914,55 @@ get_attrdef_oid(Oid relationId, AttrNumber attnum)
 }
 
 
+/*
+ * AppendAlterTableWithNextvalDefaultStmt appends a string representing either:
+ * ALTER TABLE ADD COLUMN .. DEFAULT nextval(), if TypeName isn't empty, or
+ * ALTER TABLE ALTER COLUMN .. SET DEFAULT nextval()
+ * If sequence type is not bigint, we use worker_nextval() instead of nextval().
+ */
+static void
+AppendAlterTableWithNextvalDefaultStmt(StringInfo buf, Oid sequenceOid, Oid relationId,
+									   char *colname, TypeName *typeName)
```

Comment:
I'd have a slight preference for splitting this up into 2 functions instead of changing the outcome based on typeName.